backend/amd64/selection.ml

```diff
@@ -298,53 +298,19 @@ method! select_operation op args dbg =
       self#select_floatarith true width Imulf Ifloatmul args
   | Cdivf width ->
       self#select_floatarith false width Idivf Ifloatdiv args
-  (* Special cases overriding C implementations. *)
-  | Cextcall { func = "sqrt"; alloc = false; } ->
-     begin match args with
-       [Cop(Cload { memory_chunk = Double as chunk; _}, [loc], _dbg)] ->
-         let (addr, arg) = self#select_addressing chunk loc in
-         (Ispecific(Ifloatsqrtf (Float64, addr)), [arg])
-     | [arg] ->
-         (Ispecific Simd.(Isimd (SSE2 Sqrt_scalar_f64)), [arg])
-     | _ ->
-         assert false
-    end
-  | Cextcall { func = "caml_int64_bits_of_float_unboxed"; alloc = false;
-               ty = [|Int|]; ty_args = [XFloat] } ->
-      (match args with
-        | [Cop(Cload { memory_chunk = Double; mutability = mut; is_atomic }, [loc], _dbg)] ->
-        let c = Word_int in
-        let (addr, arg) = self#select_addressing c loc in
-        Iload { memory_chunk = c;
-                addressing_mode = addr;
-                mutability = mut;
-                is_atomic; }, [arg]
-      | _ -> Imove, args)
-  | Cextcall { func = "caml_int64_float_of_bits_unboxed"; alloc = false;
-               ty = [|Float|]; ty_args = [XInt64] } ->
-      (match args with
-      | [Cop(Cload { memory_chunk = Word_int; mutability = mut; is_atomic }, [loc], _dbg)] ->
-        let c = Double in
-        let (addr, arg) = self#select_addressing c loc in
-        Iload { memory_chunk = c;
-                addressing_mode = addr;
-                mutability = mut;
-                is_atomic; }, [arg]
-      | _ -> Imove, args)
+  (* Special cases overriding C implementations (regardless of [@@builtin]). *)
+  | Cextcall { func = ("sqrt" as func); _ }
+  | Cextcall { func = ("caml_int64_bits_of_float_unboxed" as func); _ }
+  | Cextcall { func = ("caml_int64_float_of_bits_unboxed" as func); _ }
   (* x86 intrinsics ([@@builtin]) *)
-  (* CR mslater: (float32) casting/sqrt intrinsics *)
-  | Cextcall { func; builtin = true; ty = ret; ty_args = _; } ->
-      begin match func, ret with
-      | "caml_rdtsc_unboxed", [|Int|] -> Ispecific Irdtsc, args
-      | "caml_rdpmc_unboxed", [|Int|] -> Ispecific Irdpmc, args
-      | "caml_pause_hint", ([|Val|] | [| |]) ->
-         Ispecific Ipause, args
-      | "caml_load_fence", ([|Val|] | [| |]) ->
-         Ispecific Ilfence, args
-      | "caml_store_fence", ([|Val|] | [| |]) ->
-         Ispecific Isfence, args
-      | "caml_memory_fence", ([|Val|] | [| |]) ->
-         Ispecific Imfence, args
+  | Cextcall { func; builtin = true; _ } ->
+      begin match func with
+      | "caml_rdtsc_unboxed" -> Ispecific Irdtsc, args
```

Comment:
Why are we no longer matching on the
return type? I am not necessarily
implying it was useful to do so - I am
asking mostly out of curiosity.

Reply:
No particular reason; it just made the or pattern cleaner (and we don't check it elsewhere anyway)

Reply:
According to `git blame`, @gretay-js added
these check; Greta are you OK with them
being deleted?

Reply:
I'll check with Greta, merging for now

Reply:
This was added after early versions of intrinsics had mismatches between the string name of the intrinsic and its intended meaning. It's really easy to get the names wrong or miss `unboxed` annotation on the `external` declarations. Especially when input and output register types differ, as in `float_of_bits` and back above. We added this check as a way to catch some such mismatches. It's a shame that we didn't keep it up as more intrinsics were added but if it's the only place remaining, it's OK to delete. Can we add a TODO somewhere to consider adding such checks.

Reply:
Yeah, this is the only place. I'll add a CR to reconsider this in simd_selection